core/src/main/java/org/apache/brooklyn/core/objs/proxy/InternalEntityFactory.java

```diff
@@ -465,12 +487,30 @@ public void run() {
         }
         return super.constructOldStyle(clazz, flags);
     }
-    
+
     private <T extends Entity> Class<? extends T> getImplementedBy(EntitySpec<T> spec) {
         if (spec.getImplementation() != null) {
             return spec.getImplementation();
         } else {
             return entityTypeRegistry.getImplementedBy(spec.getType());
         }
     }
+
+    private List<EntityInitializer> getDefaultInitializers() {
+        return Arrays.stream(managementContext.getConfig().getConfig(DEFAULT_INITIALIZERS_CLASSNAMES).split(","))
+                .filter(Strings::isNonEmpty)
+                .map(className -> {
+                    try {
```

Comment:
there is a canonical way we do this -- let me dig this up

Reply:
to load a java class, use `JavaBrooklynClassLoadingContext.create(mgmt).tryLoadClass(className)` -- has slightly better classpath semantics

also i wonder whether we should try the type registry first?

(longer term i'd like to clean up and standardize better how we instantiate things!)

Reply:
Done!

Reply:
What I meant is that I used the canonical way to load the class and changed this order: the code now checks the `TypeRegistry` first, then tries to load from the class loader. Thank you BTW for the bit of code you shared above 👍